source/extensions/common/wasm/wasm_vm.cc

```diff
@@ -19,7 +19,25 @@ namespace Extensions {
 namespace Common {
 namespace Wasm {
 
-void EnvoyWasmVmIntegration::error(absl::string_view message) { ENVOY_LOG(trace, message); }
+proxy_wasm::LogLevel EnvoyWasmVmIntegration::getLogLevel() {
+  switch (ENVOY_LOGGER().level()) {
+  case spdlog::level::trace:
+    return proxy_wasm::LogLevel::trace;
+  case spdlog::level::debug:
+    return proxy_wasm::LogLevel::debug;
+  case spdlog::level::info:
+    return proxy_wasm::LogLevel::info;
+  case spdlog::level::warn:
+    return proxy_wasm::LogLevel::warn;
+  case spdlog::level::err:
+    return proxy_wasm::LogLevel::error;
+  default:
+    return proxy_wasm::LogLevel::critical;
+  }
+}
+
+void EnvoyWasmVmIntegration::error(absl::string_view message) { ENVOY_LOG(error, message); }
```

Comment:
Note: This was previously logged at the `trace` level, since Envoy generally doesn't log request errors at the `error` level.

Reply:
thanks, I was not sure why this was logged at trace but now it's clear. looks like integration()->error is only called during VM initialization so that's fine. Do you want me to add that comment to the code?

Reply:
It's also called as part of `fail()`, so anytime there is crash in Proxy-Wasm module, etc.

I think logging it at the `error` level is fine, but I'll leave that decision to Envoy maintainers.

Reply:
> It's also called as part of fail()

right. let's wait for @lizan to review

Reply:
In general we shouldn't log anything at higher level than debug in request path/request data unless Envoy would crash, so for `fail()`path it should be debug level.

Reply:
hmm, `fail` is called in request path only when Wasm VM crashes (trap instruction is executed) and it stop running 
https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/master/src/v8/v8.cc#L771

If I were an Wasm extension developer, then I would definitely want to have the backtrace (emitted by `fail`) and investigate what's going on since all incoming streams are immediately closed which is the urgent issue.

But I understand the point of not debug/trace logging unless Envoy crashes.

Reply:
OK seems that log doesn't log any request data itself but just where it fails, so this should be good to go.

Reply:
cool!  :100: